packages/forms/src/form_builder.ts

```diff
@@ -38,26 +38,32 @@ function isFormControlOptions(options: FormControlOptions|{[key: string]: any}|n
  */
 export type ControlConfig<T> = [T|FormControlState<T>, (ValidatorFn|(ValidatorFn[]))?, (AsyncValidatorFn|AsyncValidatorFn[])?];
 
-// Disable clang-format to produce clearer formatting for these multiline types.
+/**
+ * ɵMaybeNullable returns T if N is false, and T|null if N is true.
+ */
+export type ɵMaybeNullable<T, N> = N extends true ? T|null : T;
```

Comment:
We could theoretically avoid the conditional type here:
```suggestion
export type ɵMaybeNullable<T, N extends null> = T|N;
```
then we either use `ɵMaybeNullable<T, null>` or `ɵMaybeNullable<T, never>` for nullable and un-nullable conditions.